### PR TITLE
Make `dvv_enabled` a bucket property

### PR DESCRIPTION
**Pull request by russelldb**
Set it's default to false on legacy/default buckets
Set it's default to true on typed buckets

Update all the tests and associated places this change touches

### CODE REVIEW

**Comment by russelldb** [comment]
Sorry @cmeiklejohn do not have the time/patience to put the optimisations and bucket prop stuff as separate things. Feel free to do it yourself, though.

**Comment by cmeiklejohn** [comment]
Do any of the tests specifically test the false case?  The case where the bucket is mecked to return [] and [dvv_enabled] both end up executing the true case, since the default of the get_value is true.

**Comment by cmeiklejohn** [comment]
Other than that, I'm :+1: on this changeset.

**Comment by russelldb** [comment]
I didn't write new tests, no. I can write some on Monday, or you can add some if you like.

**Comment by cmeiklejohn** [comment]
Cool, just wanted to make sure you didn't think something was tested that wasn't.  I'd say merge.

**Comment by russelldb** [comment]
Oh, it needs https://github.com/basho/riak_core/pull/568

File: src/riak_object.erl [hunk 1]
```diff
@@ -294,6 +295,41 @@ merge_contents(NewObject, OldObject, true) ->
     riak_kv_crdt:log_merge_errors(Bucket, Key, CRDT, Error),
     merge_acc_to_contents(MergeAcc).
 
+%% Optimisation. To save converting every meta dict to a list sorting,
+%% comapring, and coverting back again, we use this optimisation, that
```
Comment (jplock) [inline src/riak_object.erl:299]: comparing?

Comment (russelldb) [inline src/riak_object.erl:299]: Late to the game, but well spotted. New branch and PR? Or live with it like "referer"?

Comment (jplock) [inline src/riak_object.erl:299]: https://github.com/basho/riak_kv/pull/892

**Pull request merged**